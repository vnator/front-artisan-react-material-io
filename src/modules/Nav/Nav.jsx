import React from 'react';

import { makeStyles } from '@material-ui/core/styles';

import {
  Drawer,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Divider,
  Typography,
} from '@material-ui/core';

import { GroupOutlined, HomeOutlined, AppsOutlined } from '@material-ui/icons';

import { useHistory } from 'react-router-dom';
import { useIntl } from 'react-intl';

import { MAIN_ROUTES } from '../../const/routes';
import { useSelector, useDispatch } from 'react-redux';
import { setNavActive } from './redux/actions';

const useStyles = makeStyles({
  list: {
    width: 250,
  },
});

const navTo = (push, after) => to => {
  push(to);
  after();
};

const Nav = () => {
  const { formatMessage } = useIntl();
  const history = useHistory();
  const classes = useStyles();
  const dispatch = useDispatch();
  const active = useSelector(({ nav }) => nav.active);

  const close = () => dispatch(setNavActive(false));

  const goTo = navTo(history.push, close);

  return (
    <Drawer anchor="left" open={active} onClose={close}>
      <div className={classes.list}>
        <List>
          <ListItem>
            <ListItemText>
              <Typography color="primary" variant="h6">
                {formatMessage({
                  id: 'nav.title',
                })}
              </Typography>
            </ListItemText>
          </ListItem>
          <Divider />

          <ListItem button onClick={() => goTo(MAIN_ROUTES.MAIN)}>
            <ListItemIcon>
              <AppsOutlined color="primary" />
            </ListItemIcon>
            <ListItemText
              primary={formatMessage({
                id: 'nav.counter',
              })}
            />
          </ListItem>
          <Divider />

          <ListItem button onClick={() => goTo(MAIN_ROUTES.ADDRESS)}>
            <ListItemIcon>
              <HomeOutlined color="primary" />
            </ListItemIcon>
            <ListItemText
              primary={formatMessage({
                id: 'nav.address',
              })}
            />
          </ListItem>
          <Divider />

          <ListItem button onClick={() => goTo(MAIN_ROUTES.USER_LIST)}>
            <ListItemIcon>
              <GroupOutlined color="primary" />
            </ListItemIcon>
            <ListItemText
              primary={formatMessage({
                id: 'nav.users',
              })}
            />
          </ListItem>
          <Divider />
        </List>
      </div>
    </Drawer>
  );
};

export { Nav, navTo };
