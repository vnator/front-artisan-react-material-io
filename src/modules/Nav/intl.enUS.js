const nav_enUS = {
  title: 'Module Name',
  counter: 'counter',
  address: 'address',
  users: 'users',
};

export { nav_enUS };
