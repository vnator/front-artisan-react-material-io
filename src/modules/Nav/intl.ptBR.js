const nav_ptBR = {
  title: 'Module Name',
  counter: 'counter',
  address: 'address',
  users: 'users',
};

export { nav_ptBR };
