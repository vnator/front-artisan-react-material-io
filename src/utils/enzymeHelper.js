import { IntlProvider, createIntl } from 'react-intl';
import { mount, shallow } from 'enzyme';
import { flattenMessages } from '../config/flattenMessages';
import { messages } from '../messages';

// You can pass your messages to the IntlProvider. Optional: remove if unneeded.

const defaultLocale = 'pt-BR';
const locale = defaultLocale;
const wrapMessages = flattenMessages(messages[locale]);

function mountWithIntl(node) {
  return mount(node, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: {
      locale,
      defaultLocale,
      messages: wrapMessages,
    },
  });
}

function shallowWithIntl(node) {
  return shallow(node, {
    wrappingComponent: IntlProvider,
    wrappingComponentProps: {
      locale,
      defaultLocale,
      messages: wrapMessages,
    },
  });
}

const intl = createIntl({
  locale: 'pt-BR',
  defaultLocale: 'pt-BR',
  messages: wrapMessages,
});

export { mountWithIntl, shallowWithIntl, intl };
