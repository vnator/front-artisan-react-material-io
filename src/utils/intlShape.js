import Type from 'prop-types';

const intlConfig = {
  locale: Type.string,
  timeZone: Type.string,
  messages: Type.objectOf(Type.string),
  defaultLocale: Type.string,
};

const intlFormatters = {
  formatDate: Type.func,
  formatTime: Type.func,
  formatRelativeTime: Type.func,
  formatNumber: Type.func,
  formatPlural: Type.func,
  formatMessage: Type.func,
  formatHTMLMessage: Type.func,
};

const intlShape = {
  ...intlConfig,
  ...intlFormatters,
};

export { intlShape, intlConfig, intlFormatters };
