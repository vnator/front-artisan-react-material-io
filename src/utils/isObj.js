const isObj = obj => typeof obj === 'object' && !(obj instanceof Array);

export { isObj };
