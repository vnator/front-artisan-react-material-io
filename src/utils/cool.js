import { isObj } from './isObj';

const cool = (a) =>
  Object.freeze(
    Object.entries(a).reduce(
      (total, [key, val]) =>
        isObj(val)
          ? Object.freeze({ ...total, [key]: cool(val) })
          : Object.freeze({ ...total, [key]: val, isFrozen: true }),
      Object.freeze(),
    ),
  );

export { cool };
