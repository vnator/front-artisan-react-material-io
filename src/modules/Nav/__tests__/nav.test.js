import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { spy } from 'sinon';

import { Nav, navTo } from '../Nav';
import { mountWithIntl } from '../../../utils/enzymeHelper';

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore([thunk]);

describe('Nav', () => {
  it('Snapshot', () => {
    const store = mockStore({
      nav: { active: false },
    });

    const header = mountWithIntl(
      <Provider store={store}>
        <MemoryRouter>
          <Nav />
        </MemoryRouter>
      </Provider>,
    );
    expect(header).toMatchSnapshot();
  });
});

describe('navTo', () => {
  it('redirect with navTo Clojure', () => {
    const push = spy();
    const close = spy();

    const path = 'any-path';
    const goTo = navTo(push, close);

    goTo(path);

    expect(push.calledWith(path)).toBeTruthy();
    expect(close.calledOnce).toBeTruthy();
  });
});
