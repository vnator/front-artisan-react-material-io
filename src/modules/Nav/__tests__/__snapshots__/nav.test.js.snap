// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Snapshot 1`] = `ReactWrapper {}`;
